src/Chapter3.hs

```diff
@@ -971,6 +1072,24 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday deriving (Show, Ord, Eq, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend = flip elem [Saturday, Sunday]
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day = toEnum $ mod (fromEnum day + 1) 7
```

Comment:
Wow, great solution with `Enum` 💯 
Alternatively, you could use `Bounded` as well, but your solution is great 👍🏼